easybuild/easyblocks/l/lammps.py

```diff
@@ -117,37 +214,51 @@ def prepare_step(self, *args, **kwargs):
+            if self.cfg['user_packages']:
+                self.cfg['user_packages'] = [x for x in self.cfg['user_packages'] if x != 'VTK']
+            # In "recent versions" of LAMMPS there is no distinction
+            self.cfg['general_packages'] = [x for x in self.cfg['general_packages'] if x != 'VTK']
         if not get_software_root('ScaFaCoS'):
-            self.cfg['user_packages'] = [x for x in self.cfg['user_packages'] if x != 'SCAFACOS']
-        if not get_software_root('yaff'):
-            self.cfg['user_packages'] = [x for x in self.cfg['user_packages'] if x != 'YAFF']
-            self.cfg['sanity_check_commands'] = [x for x in self.cfg['sanity_check_commands']
-                                                 if 'yaff' not in x]
+            if self.cfg['user_packages']:
+                self.cfg['user_packages'] = [x for x in self.cfg['user_packages'] if x != 'SCAFACOS']
+            # In "recent versions" of LAMMPS there is no distinction
+            self.cfg['general_packages'] = [x for x in self.cfg['general_packages'] if x != 'SCAFACOS']
 
-        cuda = get_software_root('CUDA')
         # list of CUDA compute capabilities to use can be specifed in two ways (where (2) overrules (1)):
         # (1) in the easyconfig file, via the custom cuda_compute_capabilities;
         # (2) in the EasyBuild configuration, via --cuda-compute-capabilities configuration option;
         ec_cuda_cc = self.cfg['cuda_compute_capabilities']
         cfg_cuda_cc = build_option('cuda_compute_capabilities')
-        cuda_cc = check_cuda_compute_capabilities(cfg_cuda_cc, ec_cuda_cc)
+        if cfg_cuda_cc and not isinstance(cfg_cuda_cc, list):
+            raise EasyBuildError("cuda_compute_capabilities in easyconfig should be provided as list of strings, " +
+                                 "(for example ['8.0', '7.5']). Got %s" % cfg_cuda_cc)
+        cuda_cc = check_cuda_compute_capabilities(cfg_cuda_cc, ec_cuda_cc, cuda=self.cuda)
 
         # cmake has its own folder
         self.cfg['srcdir'] = os.path.join(self.start_dir, 'cmake')
 
         # Enable following packages, if not configured in easyconfig
-        default_options = ['BUILD_DOC', 'BUILD_EXE', 'BUILD_LIB', 'BUILD_TOOLS']
+        if LooseVersion(self.cur_version) >= LooseVersion(self.ref_version):
+            default_options = ['BUILD_TOOLS']
+        else:
+            default_options = ['BUILD_DOC', 'BUILD_EXE', 'BUILD_LIB', 'BUILD_TOOLS']
+
         for option in default_options:
             if "-D%s=" % option not in self.cfg['configopts']:
                 self.cfg.update('configopts', '-D%s=on' % option)
 
+        # For "recent" versions, don't build docs by default (as they use a venv and pull in deps)
+        if LooseVersion(self.cur_version) >= LooseVersion(self.ref_version):
+            if "-DBUILD_DOC=" not in self.cfg['configopts']:
+                self.cfg.update('configopts', '-DBUILD_DOC=off')
```

Comment:
```suggestion
        # don't build docs by default (as they use a venv and pull in deps)
        if "-DBUILD_DOC=" not in self.cfg['configopts']:
            self.cfg.update('configopts', '-DBUILD_DOC=off')
```

Since we see issues building the docs in older versions (https://github.com/easybuilders/easybuild-easyconfigs/issues/12211) I suggest that we remove the version check here.